### PR TITLE
SCMOD-7829: Restrict management port to access swagger page

### PR DESCRIPTION
**Pull request by thilagasanthosh**

### CODE REVIEW

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/HealthcheckInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.services.staging;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+public class HealthcheckInterceptor extends HandlerInterceptorAdapter
```
Comment (Anthony-McGreevy) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/HealthcheckInterceptor.java]: make class final

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/HealthcheckInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+{
+    private final int adminPort;
+
+    public HealthcheckInterceptor(final int adminPort)
+    {
+        this.adminPort = adminPort;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception
```
Comment (Anthony-McGreevy) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/HealthcheckInterceptor.java]: make params final

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/SwaggerInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.services.staging;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+public class SwaggerInterceptor extends HandlerInterceptorAdapter
```
Comment (Anthony-McGreevy) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/SwaggerInterceptor.java]: make class final

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/SwaggerInterceptor.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+{
+    private final int adminPort;
+
+    public SwaggerInterceptor(final int adminPort)
+    {
+        this.adminPort = adminPort;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception
```
Comment (Anthony-McGreevy) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/SwaggerInterceptor.java]: make params final

**Review by Anthony-McGreevy — CHANGES_REQUESTED** [review]

**Review by Anthony-McGreevy — APPROVED** [review]
Looks good, just update the release notes like [here](https://github.houston.softwaregrp.net/caf/grammar-service/blob/master/release-notes-2.2.0.md).

**Comment by thilagasanthosh** [comment]
Updated release notes

**Pull request merged**